Fav feature

dimi-23 opened the pull request:
**Fixes issue:**
<!-- [Mention the issue number it fixes or add the details of the changes if it doesn't has a specific issue. -->
#14

**Changes:**
<!-- Add here what changes were made in this awesome pull request. -->
Added favorites feature.

(1) While searching the stars appear in front of the file names by clicking on which the user can favorite/unfavorite a file: 
![image](https://user-images.githubusercontent.com/28829807/36074120-8ae8e88a-0f61-11e8-8a2c-32fd7933e588.png)

(2) The favorited files appear at the bottom of the start-up page of the extension:
![image](https://user-images.githubusercontent.com/28829807/36074112-742111fe-0f61-11e8-8b98-485451149a8a.png)

<!-- Make sure that you enjoyed being the part of the OpenGenus Community. We would love to hear how can we make your contributing experience better. Thank you. Have a nice day! -->

AdiChat commented on code/popup.html:
```diff
@@ -6,7 +6,22 @@
   <link rel="shortcut icon" type="image/png" href="/icon/icon.png"/>
-  <title>OpenGenus: World's first Offline Search Engine</title>
+  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
+  <style>
```
Move styling to its own file 👍

AdiChat commented on code/js/popup.js:
```diff
@@ -14,6 +15,24 @@ function AddNewTags (tagName)
 	tags.push(tagName);
 }
 
+function updateFavs(x, filename) {
```
Local storage will be deleted when the user will clear his/ her browsing history with cookies and other site data option selected. 

An ideal user may clear history occasionally but we do not want to lose all the data. 

So, you may explore other options like saving in a file in appropriate format.

AdiChat commented:
Great work 😄 

I have added a few comments to aid you to make this feature the best. Kindly review them and make corresponding changes. 👍

dimi-23 commented:
Hi @AdiChat!

I've made the following  changes as mentioned by you:
- moved styling into its own file
- used Chrome Storage instead of local storage. It solves the problem that you mentioned by using chrome.storage.sync. The stored data will automatically be synced to any Chrome browser that the user is logged into, provided the user has sync enabled. When Chrome is offline, Chrome stores the data locally. The next time the browser is online, Chrome syncs the data. Even if a user disables syncing, storage.sync will still work. In this case, it will behave identically to storage.local.
- Also, since inline functions are no more accepted by Chrome Extensions, replaced them by other functions for the dynamically created elements.

AdiChat commented:
You must move the javascript and css files to the `js` and `css` folders respectively. 

A lot of files have been included along with the minified versions. You must keep only those that are being used. We must keep the size minimum and not include anything extra. You may, also, edit fontawesome css and keep only the parts that is required. 
You must, also, delete the files `popup_BACKUP_8616.html`, `popup.js.orig`, binary files like `fa-regular-400.woff` and the like.
In short, you must remove all unused files. 👍

dimi-23 commented:
Sorry, I forgot to save space and pushed the repo carelessly. Have made the changes now, you can check them!

dimi-23 commented:
I have made the following changes:

- Corrected the position of stars
- Created a favorites button which opens up favorites in a new page
- UI without any favorites:
![image](https://user-images.githubusercontent.com/28829807/36398641-1cff6522-15ee-11e8-8cf5-b39188f26d10.png)
- UI with favorites:
![image](https://user-images.githubusercontent.com/28829807/36398661-370a661a-15ee-11e8-8298-ace89baa3650.png)

AdiChat commented:
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

dimi-23 commented:
Yes, am loving it! Thank you so much! 💛
